lib/atlassian-jwt-authentication/verify.rb

```diff
@@ -77,15 +79,15 @@ def verify
         payload, header = decoder.decode_segments
       rescue JWT::VerificationError
         log(:error, "Error decoding JWT segments - signature is invalid")
-        return false
+        return UNVERIFIED_RESULT
       rescue JWT::ExpiredSignature
         log(:error, "Error decoding JWT segments - signature is expired at #{data['exp']}")
-        return false
+        return UNVERIFIED_RESULT
       end
 
       unless header && payload
         log(:error, "Error decoding JWT segments - no header and payload for client_key #{data['iss']} and addon_key #{addon_key}")
-        return false
+        return UNVERIFIED_RESULT
```

Comment:
It is golang style 😆 
Actually don't see a problem but try to raise an exception

Reply:
I can try to clean this code up, but remember why we are not raising here (and elsewhere around) exceptions:
* we have two middlewares (one for Free, one for Pro), both verifying the same request
* I reckon that was the reason we were hesitant adding exceptions originating from .verify. Otherwise clockwork free was stopping to work...

I might be wrong though.

Reply:
> I might be wrong though.

So maybe check it?

Reply:
Maybe not. My time is not unlimited (unfortunately), so I'd prefer to spend it on meaningful things instead. Tweaking the atlassian-jwt-authentication library for the sake of tweaking it is not something giving any value to our customers. Or to us to be honest.

If it works, don't touch it. Too critical component to play with.

I'll take a look here what can be improved, but I don't want to drag with this task more than 1 day. I/we've already spent more time than I've anticipated/wanted: [link](https://herocoders.atlassian.net/plugins/servlet/ac/clockwork-cloud/clockwork-timesheet?project.key=CLK&project.id=11304#!reportName=Clockwork%20Timesheet&scope%5BstartingAt%5D=2022-06-08&scope%5BendingAt%5D=2022-06-22&selectedBreakdowns%5B%5D=users&selectedBreakdowns%5B%5D=issues&selectedFilters%5BissueKeys%5D%5B%5D=CLK-1503&period=PERIOD_MONTH)
![image](https://user-images.githubusercontent.com/62769721/175020384-4120029c-9e20-4517-b57f-5aeb3c2bfdb2.png)

Reply:
I don't insist.